xchange-huobi/src/main/java/org/knowm/xchange/huobi/HuobiAdapters.java

```diff
@@ -92,19 +88,52 @@ static ExchangeMetaData adaptToExchangeMetaData(
       pairs.put(pair, adaptPair(assetPair, pairsMetaData.getOrDefault(pair, null)));
     }
 
-    Map<Currency, CurrencyMetaData> currenciesMetaData = staticMetaData.getCurrencies();
     Map<Currency, CurrencyMetaData> currencies = new HashMap<>();
-    for (HuobiAsset asset : assets) {
-      Currency currency = adaptCurrency(asset.getAsset());
-      CurrencyMetaData metadata = currenciesMetaData.getOrDefault(currency, null);
-      BigDecimal withdrawalFee = metadata == null ? null : metadata.getWithdrawalFee();
-      int scale = metadata == null ? 8 : metadata.getScale();
-      currencies.put(currency, new CurrencyMetaData(scale, withdrawalFee));
+    for (HuobiCurrencyWrapper huobiCurrencyWrapper : currencyWrapper) {
+      boolean isDelisted = DELISTED.equals(huobiCurrencyWrapper.getInstStatus());
+      CurrencyMetaData currencyMetaData = adaptCurrencyMetaData(huobiCurrencyWrapper, isDelisted);
+      Currency currency = new Currency(huobiCurrencyWrapper.getCurrency());
+      currencies.put(currency, currencyMetaData);
     }
 
     return new ExchangeMetaData(pairs, currencies, null, null, false);
   }
 
+  private static CurrencyMetaData adaptCurrencyMetaData(HuobiCurrencyWrapper huobiCurrencyWrapper, boolean isDelisted) {
+    CurrencyMetaData result = null;
+    List<HuobiCurrency> huobiCurrencies = Arrays.asList(huobiCurrencyWrapper.getHuobiCurrencies());
+    if(!huobiCurrencies.isEmpty()) {
+      result = getCurrencyMetaData(huobiCurrencies.get(0), isDelisted);
+      for (HuobiCurrency huobiCurrency : huobiCurrencies) {
+        if(TARGET_NETWORK.equals(huobiCurrency.getDisplayName())) {
+          result = getCurrencyMetaData(huobiCurrency, isDelisted);
+          break;
+        }
+      }
+    }
+    return result;
+  }
+
+  private static CurrencyMetaData getCurrencyMetaData(HuobiCurrency huobiCurrency, boolean isDelisted) {
+    int withdrawPrecision = huobiCurrency.getWithdrawPrecision();
+    BigDecimal transactFeeWithdraw = new BigDecimal(huobiCurrency.getTransactFeeWithdraw());
+    BigDecimal minWithdrawAmt = new BigDecimal(huobiCurrency.getMinWithdrawAmt());
+    WalletHealth walletHealthStatus = isDelisted ? WalletHealth.OFFLINE : getWalletHealthStatus(huobiCurrency);
+    return new CurrencyMetaData(withdrawPrecision, transactFeeWithdraw, minWithdrawAmt, walletHealthStatus);
+  }
+
+  private static WalletHealth getWalletHealthStatus(HuobiCurrency huobiCurrency) {
+    WalletHealth walletHealth = WalletHealth.ONLINE;
```

Comment:
İlk başta Wallet durumu OFFLINE mı olmalıydı ?

Reply:
Delisted ise direk offline. Aksi durumda basta online yapip duruma gore status unu degistiriyor.

Reply:
Tamamdır aslında gözden kaçırmışım. Geç vakitte baktığım için olmuş.